carma-messenger-core/j2735_convertor/src/control_request_convertor.cpp

```diff
@@ -54,101 +90,187 @@ void convert(const j2735_msgs::ControlBounds& in_msg, cav_msgs::ControlBounds& o
 // Therefore it is safe to truncate and then use the modulo operator
 // Therefore control requests sent from this component cannot have accuracy better than 1m
 
-bool isIntegerDivisable(const std::vector<double>& data, int16_t divisor) {
-  for (auto element : data) {
+bool isIntegerDivisable(const std::vector<double>& data, int16_t divisor) 
+{
+  for (auto element : data) 
+  {
     int64_t int_element = (int64_t)element; // Cast to integer to allow use of modulo
-    if (int_element % divisor != 0) {
+    if (int_element % divisor != 0) 
+    {
       return false;
     }
   }
 
   return true;
 }
 
-bool isLessThan(const std::vector<double>& data, int16_t threshold) {
-  for (auto element : data) {
-    if (element > fabs(threshold)) {
+bool isLessThan(const std::vector<double>& data, int16_t threshold) 
+{
+  for (auto element : data) 
+  {
+    if (element > fabs(threshold))
+    {
       return false;
     }
   }
 
   return true;
 }
 
-void convert(const cav_msgs::ControlBounds& in_msg, j2735_msgs::ControlBounds& out_msg, const int8_t scale) {
+void convert(const cav_msgs::TrafficControlBounds& in_msg, j2735_msgs::TrafficControlBounds& out_msg, const int8_t scale) 
+{
   out_msg.oldest = (in_msg.oldest.sec * units::MS_PER_S) + (in_msg.oldest.nsec / units::NS_PER_MS_INT);
-  out_msg.latitude = in_msg.latitude * units::TENTH_MICRO_DEG_PER_DEG;
-  out_msg.longitude = in_msg.longitude * units::TENTH_MICRO_DEG_PER_DEG;
-  
-  double scaled_x = in_msg.offsets[0] / pow(10, scale);
-  double scaled_y = in_msg.offsets[1] / pow(10, scale);
-  double scaled_z = in_msg.offsets[2] / pow(10, scale);
+  out_msg.reflat = in_msg.reflat * units::TENTH_MICRO_DEG_PER_DEG;
+  out_msg.reflon = in_msg.reflon * units::TENTH_MICRO_DEG_PER_DEG;
+
+  double scaled_deltax0 = in_msg.offsets[0].deltax / pow(10, scale);
+  double scaled_deltay0 = in_msg.offsets[1].deltay / pow(10, scale);
+  double scaled_deltax1 = in_msg.offsets[0].deltax / pow(10, scale);
+  double scaled_deltay1 = in_msg.offsets[1].deltay / pow(10, scale);
+  double scaled_deltax2 = in_msg.offsets[0].deltax / pow(10, scale);
+  double scaled_deltay2 = in_msg.offsets[1].deltay / pow(10, scale);
```

Comment:
I am confused by the indices here. In my opinion, you would access them with the same order you are declaring your variable? e.g. in_msg.offsets[0].deltax ... in_msg.offsets[0].deltay... ain_msg.offsets[1].deltax ... in_msg.offsets[1.deltay etc